synapse/rest/media/v1/thumbnailer.py

```diff
@@ -129,24 +118,62 @@ def crop(self, width: int, height: int, output_type: str) -> BytesIO:
         Returns:
             BytesIO: the bytes of the encoded image ready to be written to disk
         """
+
+        return self._process_animated_aware(
+            self.image, output_type, lambda i: self._crop(i, width, height)
+        )
+
+    def _process_animated_aware(
```

Comment:
rule of three *should* apply here, but this piece is a bit complex, and the only thing that'd really change between the two functions is changing `_resize` with `_crop`, thus the abstraction